Source/Validation.swift

```diff
@@ -133,45 +119,27 @@ extension Request {
     public func validate<S: Sequence>(contentType acceptableContentTypes: S) -> Self where S.Iterator.Element == String {
         return validate { _, response in
             guard let validData = self.delegate.data, validData.count > 0 else { return .success }
-
-            if let responseContentType = response.mimeType, let responseMIMEType = MIMEType(responseContentType) {
-                for contentType in acceptableContentTypes {
-                    if let acceptableMIMEType = MIMEType(contentType), acceptableMIMEType.matches(responseMIMEType) {
-                        return .success
-                    }
-                }
-            } else {
+            
+            guard
+                let responseContentType = response.mimeType,
+                let responseMIMEType = MIMEType(responseContentType) else {
                 for contentType in acceptableContentTypes {
                     if let mimeType = MIMEType(contentType), mimeType.type == "*" && mimeType.subtype == "*" {
                         return .success
                     }
                 }
+                    
+                return .failure(AFError.responseValidationFailed(reason: .contentTypeMissingAndNotWildcard))
             }
 
-            let contentType: String
-            let failureReason: String
-
-            if let responseContentType = response.mimeType {
-                contentType = responseContentType
-
-                failureReason = (
-                    "Response content type \"\(responseContentType)\" does not match any acceptable " +
-                    "content types: \(acceptableContentTypes)"
-                )
-            } else {
-                contentType = ""
-                failureReason = "Response content type was missing and acceptable content type does not match \"*/*\""
+            for contentType in acceptableContentTypes {
+                if let acceptableMIMEType = MIMEType(contentType), acceptableMIMEType.matches(responseMIMEType) {
+                    return .success
+                }
             }
-
-            let error = NSError(
-                domain: ErrorDomain,
-                code: ErrorCode.contentTypeValidationFailed.rawValue,
-                userInfo: [
-                    NSLocalizedFailureReasonErrorKey: failureReason,
-                    ErrorUserInfoKeys.ContentType: contentType
-                ]
-            )
-
+            
+            let error = AFError.responseValidationFailed(reason: .contentTypeMismatch(acceptable: acceptableContentTypes as! [String], actual: responseContentType))
```

Comment:
This is long enough, can we break this up into a couple lines? Also, is the `as!` guaranteed to be safe here?